packages/firebase_ml_vision/lib/src/text_detector.dart

```diff
@@ -31,39 +31,53 @@ class TextDetector implements FirebaseVisionDetector {
 
     return blocks;
   }
+
+  /// Closes the text detector and release its model resources.
+  @override
+  Future<void> close() {
+    return FirebaseVision.channel.invokeMethod('TextDetector#close');
+  }
 }
 
 /// Abstract class representing dimensions of recognized text in an image.
 abstract class TextContainer {
   TextContainer._(Map<dynamic, dynamic> data)
-      : boundingBox = Rectangle<num>(
-          data['left'],
-          data['top'],
-          data['width'],
-          data['height'],
-        ),
-        cornerPoints = data['points']
+      : boundingBox = data['left'] != null
+            ? Rectangle<num>(
+                data['left'],
+                data['top'],
+                data['width'],
+                data['height'],
+              )
+            : null,
+        _cornerPoints = data['points']
             .map<Point<num>>((dynamic item) => Point<num>(
                   item[0],
                   item[1],
                 ))
             .toList(),
         text = data['text'];
 
-  /// Axis-aligned bounding rectangle of the detected text.
-  final Rectangle<num> boundingBox;
+  final List<Point<num>> _cornerPoints;
 
-  /// The four corner points in clockwise direction starting with top-left.
+  /// Axis-aligned bounding rectangle of the detected text.
   ///
-  /// Due to the possible perspective distortions, this is not necessarily a
-  /// rectangle. Parts of the region could be outside of the image.
-  final List<Point<num>> cornerPoints;
+  /// Could be null even if text is found.
+  final Rectangle<num> boundingBox;
 
   /// The recognized text as a string.
   ///
   /// Returned in reading order for the language. For Latin, this is top to
   /// bottom within a Block, and left-to-right within a Line.
   final String text;
+
+  /// The four corner points in clockwise direction starting with top-left.
+  ///
+  /// Due to the possible perspective distortions, this is not necessarily a
+  /// rectangle. Parts of the region could be outside of the image.
+  ///
+  /// Could be empty even if text is found.
```

Comment:
Any idea what that means or when that happens?

Reply:
The android implementation has the `@Nullable` annotation for `boundingBox()` and `cornerPoints()`. I haven't found anywhere that specifies why or had it actually return `null` for any of the test images. However, I added code and a comment to handle it just in case it does because I didn't want it to accidentally crash an app. Also, I get warnings from android studio to handle a `null` situation.

What's interesting though, is that the iOS side marks `cornerPoints()` as `_NonNull`. 
https://firebase.google.com/docs/reference/ios/firebasemlvision/api/reference/Protocols/FIRVisionText